components/automate-cli/pkg/verifyserver/services/opensearchbackupservice/opensearchs3backupmanaged.go

```diff
@@ -0,0 +1,183 @@
+package opensearchbackupservice
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/lib/logger"
+	"github.com/gofiber/fiber"
+	elastic "github.com/olivere/elastic/v7"
+	elasticaws "github.com/olivere/elastic/v7/aws/v4"
+	"github.com/pkg/errors"
+)
+
+type IOSS3BackupService interface {
+	OSS3BackupVerify(models.S3BackupDetails, *fiber.Ctx) (models.S3BackupManagedResponse, error)
+}
+
+type IOpenSearchclient interface {
+	CreateAWSClient(models.S3BackupDetails, string) (*elastic.Client, error)
+}
+
+type OSS3BackupService struct {
+	OSClient     IOpenSearchclient
+	OSOperations IOpensearchOperations
+	Log          logger.Logger
+}
+
+type SnapshotRepoRequest struct {
+	Type     string                 `json:"type"`
+	Settings map[string]interface{} `json:"settings"`
+}
+
+type SnapshotRequest struct {
+	Indices           string `json:"indices"`
+	IgnoreUnavailable bool   `json:"ignore_unavailable"`
+	AllowNoIndices    bool   `json:"allow_no_indices"`
+}
+
+type OpenSearchclient struct{}
+
+func NewOpenSearchclient() IOpenSearchclient {
+	return &OpenSearchclient{}
+}
+
+func NewOSS3BackupService(log logger.Logger) IOSS3BackupService {
+	return &OSS3BackupService{
+		OSClient:     NewOpenSearchclient(),
+		OSOperations: NewOpensearchOperations(log),
+		Log:          log,
+	}
+}
+
+func (ss *OSS3BackupService) OSS3BackupVerify(request models.S3BackupDetails, ctx *fiber.Ctx) (models.S3BackupManagedResponse, error) {
+
+	url := request.Endpoint
+
+	client, err := ss.OSClient.CreateAWSClient(request, url)
+
+	if err != nil {
+		ss.Log.Error(err)
+		return models.S3BackupManagedResponse{}, err
+	}
+
+	if _, err = ss.OSOperations.CreateTestIndex(client, ctx, TEST_INDEX_NAME); err != nil {
+		ss.Log.Error("Index creation failed: ", err)
+		return models.S3BackupManagedResponse{
+			Passed: false,
+			Checks: []models.S3BackupChecks{createFailedResponse(INDEX_CREATE_FAILED_MESSAGE, INDEX_CREATE_FAILED_RESOLUTION)},
+		}, err
+	}
+
+	snapshotCreateReq := SnapshotRepoRequestS3{
+		Bucket:   request.S3Bucket,
+		BasePath: request.S3BasePath,
+		RoleArn:  request.AWSRoleArn,
+		Region:   request.AWSRegion,
+	}
+
+	if _, err = ss.OSOperations.CreateSnapshotRepo(client, ctx, snapshotCreateReq, TEST_REPO_NAME); err != nil {
```

Comment:
Could you add some debug logs after each step for us to know execution incase of any issues

Reply:
done